### PR TITLE
File search extension

### PR DESCRIPTION
**Pull request by SeraphimaZykova**
## PR Checklist

- [ ] **Closes:** #xxx
- [ ] **Communication:** I've discussed this with core contributors already. If work hasn't been agreed, this work might be rejected
- [ ] **Tests:** Added/updated and all pass
- [ ] **Localization:** All end user facing strings can be localized
- [ ] **Dev docs:** Added/updated
- [ ] **New binaries:** Added on the required places
   - [ ] [JSON for signing](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ESRPSigning_core.json) for new binaries
   - [ ] [WXS for installer](https://github.com/microsoft/PowerToys/blob/main/installer/PowerToysSetup/Product.wxs) for new binaries and localization folder
   - [ ] [YML for CI pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ci/templates/build-powertoys-steps.yml) for new test projects
   - [ ] [YML for signed pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/release.yml)
- [ ] **Documentation updated:** If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/windows-uwp/tree/docs/hub/powertoys) and link it here: #xxx

### CODE REVIEW

File: src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.Indexer/Indexer/SearchQueryBase.cs [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+                null);
+
+            if (res != 0)
+            {
+                Logger.LogError($"Error releasing rows: {res}");
+                break;
+            }
+
+            // Free the memory allocated for row handles
+            Marshal.FreeCoTaskMem(prghRows);
```
Comment (zadjii-msft) [inline src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.Indexer/Indexer/SearchQueryBase.cs:101]: This seems to be the line that's throwing the heap corruption. Yea of course just commenting it out _works_, but leaks memory. I wonder why this is throwing - `prghRows` looked like it had a valid non-zero value so I don't what's up here

**Comment by zadjii-msft** [comment]
I tossed some thoughts in `dev/migrie/indexer`. Nothing major, just 

* f5daf4078 Try to do it even more async, but of course, with quite a bit more crashing
* 6bba35776 make it a builtin
* 28a2e117f small polish: folder icons, placeholder text
* (as well as a merge with a branch that's got some other UX updates)

**Review by zadjii-msft — COMMENTED** [review]
Okay there's 
* a couple places where there's the side-effects of a bad merge with main, and
* A number of files that were added when this was a out-of-proc extension, rather than in-proc that can be deleted

but otherwise, I think this is probably fine to merge

File: src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/Messages/UpdateActionBarPage.cs
[the remark this answers is not in the document]
Comment (SeraphimaZykova) [reply src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/Messages/UpdateActionBarPage.cs:1]: Thanks for reviewing! I removed all changes unrelated to the Indexer that appeared after a bad merge.

File: src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.Indexer/Indexer/OleDB/Constants.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.CmdPal.Ext.Indexer.Indexer.OleDB;
+
+public static class Constants
+{
+    public static readonly Guid DBGUIDDEFAULT = new("C8B521FB-5CF3-11CE-ADE5-00AA0044773D");
```
[the remark this answers is not in the document]
Comment (SeraphimaZykova) [reply src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.Indexer/Indexer/OleDB/Constants.cs]: It's not being used. I've cleaned it up. Thanks!

**Review by zadjii-msft — APPROVED** [review]
this is nice

**Pull request merged**